core/lib/Funext.agda

```diff
@@ -135,29 +137,222 @@ module StrongFunextDep {j} {P : A → Type j} where
+    app=-β (λ a' → α a' ∙ β a') a ◃∎
+      =ₛ₁⟨ 0 & 1 & step₄ ⟩
+    ap (λ p → app= p a) (λ=-η (λ= α ∙ λ= β)) ◃∙
+    ap (λ γ → app= (λ= γ) a)
+       (λ= (λ a' → ap-∙ (λ γ → γ a') (λ= α) (λ= β) ∙
+                   ap2 _∙_ (app=-β α a') (app=-β β a'))) ◃∙
+    app=-β (λ a' → α a' ∙ β a') a ◃∎
+      =ₛ₁⟨ 1 & 1 &
+           ap-∘ (λ p → app= p a) λ=
+                (λ= (λ a' → ap-∙ (λ γ → γ a') (λ= α) (λ= β) ∙
+                            ap2 _∙_ (app=-β α a') (app=-β β a'))) ⟩
+    ap (λ p → app= p a) (λ=-η (λ= α ∙ λ= β)) ◃∙
+    ap (λ p → app= p a)
+       (ap λ=
+           (λ= (λ a' → ap-∙ (λ γ → γ a') (λ= α) (λ= β) ∙
+                       ap2 _∙_ (app=-β α a') (app=-β β a')))) ◃∙
+    app=-β (λ a' → α a' ∙ β a') a ◃∎
+      =ₛ⟨ 0 & 2 & ∙-ap-seq (λ p → app= p a) (∙-λ=-seq α β) ⟩
+    ap (λ p → app= p a) (=ₛ-out (∙-λ= α β)) ◃∙
+    app=-β (λ a' → α a' ∙ β a') a ◃∎ ∎ₛ
+    where
+    step₄ : ! (app=-β (λ a' → app= (λ= α ∙ λ= β) a') a) ==
+            ap (λ p → app= p a) (λ=-η (λ= α ∙ λ= β))
+    step₄ =
+      ! (app=-β (λ a' → app= (λ= α ∙ λ= β) a') a)
+        =⟨ ap ! (! (app=-β (app=-β (λ a' → app= (λ= α ∙ λ= β) a')) a)) ⟩
+      ! (app= (λ= (app=-β (λ a' → app= (λ= α ∙ λ= β) a'))) a)
+        =⟨ !-ap (λ γ → γ a) (λ= (app=-β (λ a' → app= (λ= α ∙ λ= β) a'))) ⟩
+      app= (! (λ= (app=-β (λ a' → app= (λ= α ∙ λ= β) a')))) a
+        =⟨ ap (λ p → app= p a) (! (λ=-app=-adj' (λ= α ∙ λ= β))) ⟩
+      app= (ap app= (λ=-η (λ= α ∙ λ= β))) a
+        =⟨ ∘-ap (λ f → f a) app= (λ=-η (λ= α ∙ λ= β)) ⟩
+      ap (λ p → app= p a) (λ=-η (λ= α ∙ λ= β)) =∎
+
+module _ {j} {B : A → Type j} {f g h k : Π A B}
+         (α : f ∼ g) (β : g ∼ h) (γ : h ∼ k) where
+
+  ∙∙-λ= : λ= α ◃∙ λ= β ◃∙ λ= γ ◃∎ =ₛ λ= (λ x → α x ∙ β x ∙ γ x) ◃∎
+  ∙∙-λ= =
+    λ= α ◃∙ λ= β ◃∙ λ= γ ◃∎
+      =ₛ⟨ 1 & 2 & ∙-λ= β γ ⟩
+    λ= α ◃∙ λ= (λ x → β x ∙ γ x) ◃∎
+      =ₛ⟨ ∙-λ= α (λ x → β x ∙ γ x) ⟩
+    λ= (λ x → α x ∙ β x ∙ γ x) ◃∎ ∎ₛ
+
+  λ=-∙∙ : λ= (λ x → α x ∙ β x ∙ γ x) ◃∎ =ₛ λ= α ◃∙ λ= β ◃∙ λ= γ ◃∎
+  λ=-∙∙ = !ₛ ∙∙-λ=
+
+module _ {j} {B : A → Type j} {f g : Π A B} where
+
+  !-app= : (α : f == g) → λ= (! ∘ app= α) == ! α
+  !-app= idp = ! (λ=-η idp)
+
+  !-λ= : (α : f ∼ g) → λ= (! ∘ α) == ! (λ= α)
+  !-λ= α =
+    λ= (! ∘ α)
+      =⟨ ap λ= (λ= (λ a → ap ! (! (app=-β α a)))) ⟩
+    λ= (! ∘ app= (λ= α))
+      =⟨ !-app= (λ= α) ⟩
+    ! (λ= α) =∎
+
+  λ=-! : (α : f ∼ g) → ! (λ= α) == λ= (! ∘ α)
+  λ=-! α = ! (!-λ= α)
+
+module _ {j k} {B : A → Type j} {C : A → Type k}
+  {f g : Π A B} (h : (a : A) → B a → C a) where
+
+  app=-ap : (α : f == g)
+    → ap (λ f' a → h a (f' a)) α == λ= (λ a → ap (h a) (app= α a))
+  app=-ap idp = λ=-η idp
+
+  λ=-ap : (α : f ∼ g)
+    → ap (λ f' a → h a (f' a)) (λ= α) == λ= (λ a → ap (h a) (α a))
+  λ=-ap α =
+    ap (λ f' a → h a (f' a)) (λ= α)
+      =⟨ app=-ap (λ= α) ⟩
+    λ= (λ a → ap (h a) (app= (λ= α) a))
+      =⟨ ap λ= (λ= (λ a → ap (ap (h a)) (app=-β α a))) ⟩
+    λ= (λ a → ap (h a) (α a)) =∎
```

Comment:
@timjb To help me understand the change to path sequences, could you possibly point out the parts that really need PathSeq at the core kernel?